oak-jcr/src/main/java/org/apache/jackrabbit/oak/jcr/session/SessionNamespaces.java

```diff
@@ -86,6 +88,11 @@ synchronized void setNamespacePrefix(String prefix, String uri)
 
         // add the new mapping
         local.put(prefix, uri);
+
+        // make sure the previously mapped URI has a prefix
+        if (previouslyMappedUri != null) {
+            getNamespacePrefix(previouslyMappedUri);
```

Comment:
So this works because getNamespacePrefix will put put a new prefix in the local mapping, if not found.

The behavior of getNamespacePrefix is not changed (always was like that). And I assume setNamespacePrefix was not called a lot. So I think this change has a low risk.